website/server/libs/payments/paypal.js

```diff
@@ -162,6 +163,8 @@ api.checkoutSuccess = async function checkoutSuccess (options = {}) {
 
     data.paymentMethod = 'PayPal (Gift)';
     data.gift = gift;
+
+    promoSubscription(data);
```

Comment:
I feel like there could be some issues with this:

1. There's no error handling
2. The subscription is applied before the payment is done (while usually it's done later, see line 173 here)

I would maybe move the application of the promo sub to `payments/subscriptions#createSubscription` so that it also avoids having to change the code for each payment provider